packages/web3/src/web3.ts

```diff
@@ -67,14 +67,56 @@ export class Web3 extends Web3Context<EthExecutionAPI> {
 	public eth: Web3EthInterface;
 
 	public constructor(provider?: SupportedProviders<EthExecutionAPI> | string) {
+		const signTransactionWithState = async (transaction: Transaction, privateKey: Bytes) => {
+			const tx = await prepareTransactionForSigning(transaction, this);
+
+			const privateKeyBytes = format({ eth: 'bytes' }, privateKey, ETH_DATA_FORMAT);
+
+			return signTransaction(tx, privateKeyBytes);
+		};
+
+		const privateKeyToAccountWithState = (privateKey: Buffer | string) => {
+			const account = privateKeyToAccount(privateKey);
+
+			return {
+				...account,
+				signTransaction: async (transaction: Transaction) =>
+					signTransactionWithState(transaction, account.privateKey),
+			};
+		};
+
+		const decryptWithState = async (
+			keystore: string,
+			password: string,
+			options?: Record<string, unknown>,
+		) => {
+			const account = await decrypt(
+				keystore,
+				password,
+				(options?.nonStrict as boolean) ?? true,
+			);
+
+			return {
+				...account,
+				signTransaction: async (transaction: Transaction) =>
+					signTransactionWithState(transaction, account.privateKey),
+			};
+		};
+
+		const createWithState = () => {
+			const account = create();
+
+			return {
+				...account,
+				signTransaction: async (transaction: Transaction) =>
+					signTransactionWithState(transaction, account.privateKey),
+			};
+		};
+
```

Comment:
These functions are in main web3 package for avoiding eth package dep in accounts pkg, or is there any thing else?

But I think these should be moved to accounts package as: If some one want to import signing functionality directly from accounts package the behavior will be diff. 

Or if you want to keep in current way, move these functions to new file in main web3 package and document that defaults will not be filled if these functions are directly used from accounts package instead of main web3 package.

Reply:
As mentioned earlier in a discussion: 

There are two approaches to do it.

1. Make the signTransaction stateful accepting the context and handle rest 
2. Process state related stuff in the higher level object web3 or web3.eth and keep the signTransaction stateless. 

I used second approach to have minimum changes in this PR.

Reply:
ok, sounds good. Could you do changes as discussed in chat. 

- changelog update (under breaking changes) for these functions exported via web3 main package will have diff behavior as compared to exported in accounts package. ( filling defaults )

- moving these state functions in diff file in same web3 package and continue using second approach